pkg/operator/controller/dns_status_test.go

```diff
@@ -0,0 +1,379 @@
+package controller
+
+import (
+	"fmt"
+	"testing"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestDNSStatusConditions(t *testing.T) {
+	testCases := []struct {
+		description                                string
+		clusterIP                                  string
+		avail, desire                              int32
+		degradeType, progressType, availType       string
```

Comment:
Why are `degradeType`, `progressType`, and `availType` inputs? They are all the same for all test cases, and each one has only one reasonable value.